sql-plugin/src/main/scala/com/nvidia/spark/rapids/profiler.scala

```diff
@@ -0,0 +1,389 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids
+
+import java.nio.ByteBuffer
+import java.nio.channels.{Channels, WritableByteChannel}
+import java.util.concurrent.{ConcurrentHashMap, Future, ScheduledExecutorService, TimeUnit}
+
+import scala.collection.mutable
+
+import com.nvidia.spark.rapids.jni.Profiler
+import org.apache.hadoop.fs.Path
+import org.apache.hadoop.ipc.CallerContext
```

Comment:
Could we use other alternative than`CallerContext`? It was introduced since [Hadoop 2.8 and Hadoop 3.0](https://issues.apache.org/jira/browse/HDFS-9184). It may fail some cases for older Hadoop version users.